### ISSUE TITLE
Where does the 'weight' as the input of the Focal Loss come from? 

### ISSUE THREAD

**Issue by leonnardoleo** [open-mmlab/mmdetection#39]
`def sigmoid_focal_loss(pred,
                       target,
                       weight,
                       gamma=2.0,
                       alpha=0.25,
                       reduction='elementwise_mean'):
    pred_sigmoid = pred.sigmoid()
    pt = (1 - pred_sigmoid) * target + pred_sigmoid * (1 - target)
    weight = (alpha * target + (1 - alpha) * (1 - target)) * weight
    weight = weight * pt.pow(gamma)
    return F.binary_cross_entropy_with_logits(
        pred, target, weight, reduction=reduction)`
There is an input named weight of the focal loss. Could you explain what this weight is and how I can get it. Thank you very much

**Comment by hellock** [comment]
`weight` is used for masking the elements in our code. It is the same size as `pred` and is either 0 or 1.
